command/pr.go

```diff
@@ -307,13 +307,33 @@ func prView(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	pr, _, err := prFromArgs(ctx, apiClient, cmd, args)
-	if err != nil {
-		return err
+	pr, baseRepo, prError := prFromArgs(ctx, apiClient, cmd, args)
+	if prError != nil {
+		if prError, ok := prError.(*api.PRNotFoundError); ok && !prError.UsingBranch {
+			// it could be an Issue
+			issue, issueError := issueFromArg(apiClient, baseRepo, args[0])
```

Comment:
Right now, I'm not a fan of the approach where after an API error we fall back to looking up the object of a different type. Ideally, we would always run a single API request with an issue number as input and return either an issue or a PR object. I realize that this would complicate both our current query logic and return types from functions, but we could potentially do some type-switching on those return types.

@AliabbasMerchant Do you think there would be merit to this approach?

Reply:
I think that if the API provides an in-built method to handle both of them in a single call/method, it could make sense to do so, But still, printing them would definitely require separate logic for both, which could become quite cumbersome.  
And I agree with you, it could complicate our current query logic if we try to combine them.  
I tried some kind of type-switching/unifying on the return types [here](https://github.com/cli/cli/pull/1153#issuecomment-642862574), but personally I was not satisfied with the result.